build-support/githooks/pre-commit

```diff
@@ -13,48 +14,27 @@ if [[ -n "${GIT_DIR}" && "${GIT_DIR}" != /* ]]; then
   export GIT_DIR
 fi
 
-set -e
-
 MERGE_BASE="$(git_merge_base)"
 
-# When travis builds a tag, it does so in a shallow clone without main fetched, which
-# fails in pants changed.
-if git rev-parse --verify "${MERGE_BASE}" &> /dev/null; then
-  if git diff "${MERGE_BASE}" --name-only | grep '\.rs$' > /dev/null; then
-    # Clippy happens on a different Travis CI shard because of separate caching concerns.
-    # The TRAVIS env var is documented here:
-    #   https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
-    if [[ "${TRAVIS}" != "true" ]]; then
-      echo "* Running \`./cargo clippy --all\`"
-      ./cargo clippy --all || exit 1
-    fi
-    echo "* Checking formatting of Rust files"
-    ./build-support/bin/check_rust_formatting.sh || exit 1
-    echo "* Checking Rust target headers"
-    build-support/bin/check_rust_target_headers.sh || exit 1
-  fi
-
-  echo "* Checking MyPy"
-  ./pants --changed-since="${MERGE_BASE}" --changed-dependees=transitive --tag='-nolint' typecheck || exit 1
+if git diff "${MERGE_BASE}" --name-only | grep '\.rs$' > /dev/null; then
+  echo "* Running \`./cargo clippy\`"
+  ./cargo clippy || exit 1
```

Comment:
Why does an M1 shortcoming get to take away checks for everything else? You have an M1 so it should be easy to come up with the correct conditional logic to detect M1 and do less for just M1s. I have worked on brfs in the past and this was a useful check.

Reply:
Fixed